Userland/Libraries/LibJS/Runtime/Temporal/TimeZone.cpp

```diff
@@ -262,64 +262,73 @@ ThrowCompletionOr<double> parse_time_zone_offset_string(GlobalObject& global_obj
+        factor = 1;
     }
 
-    // 8. Set hours to ! ToIntegerOrInfinity(hours).
-    auto hours = *hours_part.to_uint<u8>();
+    // 7. Assert: hours is not empty.
+    VERIFY(hours.has_value());
+
+    // 8. Let hoursMV be ! ToIntegerOrInfinity(CodePointsToString(hours)).
+    auto hours_mv = *hours->to_uint<u8>();
 
-    // 9. Set minutes to ! ToIntegerOrInfinity(minutes).
-    auto minutes = *minutes_part.value_or("0"sv).to_uint<u8>();
+    // 9. Let minutesMV be ! ToIntegerOrInfinity(CodePointsToString(minutes)).
+    auto minutes_mv = *minutes.value_or("0"sv).to_uint<u8>();
 
-    // 10. Set seconds to ! ToIntegerOrInfinity(seconds).
-    auto seconds = *seconds_part.value_or("0"sv).to_uint<u8>();
+    // 10. Let secondsMV be ! ToIntegerOrInfinity(CodePointsToString(seconds)).
+    auto seconds_mv = *seconds.value_or("0"sv).to_uint<u8>();
 
-    i32 nanoseconds;
-    // 11. If fraction is not undefined, then
-    if (fraction_part.has_value()) {
-        // a. Set fraction to the string-concatenation of the previous value of fraction and the string "000000000".
-        auto fraction = String::formatted("{}000000000", *fraction_part);
+    u32 nanoseconds_mv;
 
-        // b. Let nanoseconds be the String value equal to the substring of fraction from 1 to 10.
-        // NOTE: parse_time_zone_numeric_utc_offset_syntax(), which we use to capture TimeZoneUTCOffsetFraction, doesn't include the decimal separator.
+    // 11. If fSeconds is not empty, then
+    if (fraction.has_value()) {
+        // a. Let fSecondsDigits be the substring of CodePointsToString(fSeconds) from 1.
+        auto fraction_digits = fraction->substring_view(1);
 
-        // c. Set nanoseconds to ! ToIntegerOrInfinity(nanoseconds).
-        nanoseconds = *fraction.substring(0, 9).to_int<i32>();
+        // b. Let fSecondsDigitsExtended be the string-concatenation of fSecondsDigits and "000000000".
+        auto fraction_digits_extended = String::formatted("{}000000000", fraction_digits);
+
+        // c. Let nanosecondsDigits be the substring of fSecondsDigitsExtended from 0 to 9.
+        auto nanoseconds_digits = fraction_digits_extended.substring_view(0, 9);
+
+        // d. Let nanosecondsMV be ! ToIntegerOrInfinity(nanosecondsDigits).
+        nanoseconds_mv = *nanoseconds_digits.to_uint<u32>();
     }
     // 12. Else,
     else {
-        // a. Let nanoseconds be 0.
-        nanoseconds = 0;
+        // a. Let nanosecondsMV be 0.
+        nanoseconds_mv = 0;
     }
-    // 13. Return sign × (((hours × 60 + minutes) × 60 + seconds) × 10^9 + nanoseconds).
+
+    // 13. Return factor × (((hoursMV × 60 + minutesMV) × 60 + secondsMV) × 109 + nanosecondsMV).
     // NOTE: Decimal point in 10^9 is important, otherwise it's all integers and the result overflows!
-    return sign * (((hours * 60 + minutes) * 60 + seconds) * 1000000000.0 + nanoseconds);
+    return factor * (((hours_mv * 60 + minutes_mv) * 60 + seconds_mv) * 1000000000.0 + nanoseconds_mv);
```

Comment:
Can we just use `1e9`, I think that is a bit more readeable
Or if not that maybe some digit-seperators

Reply:
sure, thanks for the suggestion!